app/models/feed.rb

```diff
@@ -7,15 +7,28 @@ def initialize(type, account)
   end
 
   def get(limit, max_id = nil, since_id = nil)
+    if redis.exists("account:#{@account.id}:regeneration")
+      from_database(limit, max_id, since_id)
+    else
+      from_redis(limit, max_id, since_id)
+    end
+  end
+
+  private
+
+  def from_redis(limit, max_id, since_id)
     max_id     = '+inf' if max_id.blank?
     since_id   = '-inf' if since_id.blank?
     unhydrated = redis.zrevrangebyscore(key, "(#{max_id}", "(#{since_id}", limit: [0, limit], with_scores: true).map(&:last).map(&:to_i)
     status_map = Status.where(id: unhydrated).cache_ids.map { |s| [s.id, s] }.to_h
-
     unhydrated.map { |id| status_map[id] }.compact
   end
 
-  private
+  def from_database(limit, max_id, since_id)
+    results = Status.as_home_timeline(@account).paginate_by_max_id(limit, max_id, since_id).to_a
```

Comment:
> I wanted to do it this way to avoid copying the array (reject! mutates existing array afaik)

`to_a` copies an internal array, so the number of copy is same. The below is the long explanation.

Here is the implementation of `to_a`:
```Ruby
# Converts relation objects to Array.
def to_a
    records.dup
end
```
https://github.com/rails/rails/blob/v5.1.1/activerecord/lib/active_record/relation.rb#L249

`reject` is defined in `Enumerable`. `reject` itself copies the source, but there is no additional copy. The source is the very array copied in `to_a`.
> The class must provide a method `each`
https://docs.ruby-lang.org/en/2.4.0/Enumerable.html

```Ruby
    delegate :to_xml, :encode_with, :length, :each, :uniq, :to_ary, :join,
         :[], :&, :|, :+, :-, :sample, :reverse, :compact, :in_groups, :in_groups_of,
         :to_sentence, :to_formatted_s, :as_json,
         :shuffle, :split, :index, to: :records
```
https://github.com/rails/rails/blob/v5.1.1/activerecord/lib/active_record/relation/delegation.rb

In summary, both of `to_a` and `reject` are copying the same array once.